BoomBoxEx3.c

```diff
@@ -1 +1,490 @@
+  //Checks if the play button has been pressed 
+  {
+    //Serial.println(HTHLength); 
+    for (int i = 0; i < HTHLength; i++)
+    //Runs a for loop the amount of times as there are notes in HTHNotes 
+    {
+      prevPlayButton = playButtonState;
+      playButtonState = digitalRead(playButton);
+      int duration = 1000 / HTHDurations[i];
+      int delay1 = duration + 100;
+      note(8, HTHNotes[i], duration);
+      //Runs our note function in order to play a note 
+      delay(delay1);
+      lcdDo(1);
+      //Updates the LCD
+
+      if (playButtonState and !prevPlayButton)
+      //Edge conditions to make sure the user doesn't immediatly pause and then unpause the song 
+      {
+        counter2++;
+        lcdDo(4);
+        //Updates the LCD 
+        while (counter2 % 2 == 1)
+        //While loop Pauses the song 
+        {
+          prevPlayButton = playButtonState;
+          playButtonState = digitalRead(playButton);
+          if (playButtonState && !prevPlayButton) {
+            counter2++;
+            lcdDo(1);
+            break;
+            //Unpauses the song by leaving the while loop 
+          }
+        }
+        prevPlayButton = playButtonState;
+        playButtonState = digitalRead(playButton);
+      }
+    }
+  }
+} else if (songToPlay == 2) {
+  if (Update)
+    //Only updates the LCD when a new song is to be played to prevent visible updates
+  { 
+    lcdDo(2);
+    Update = !Update;
+  }
+  if (playButtonState)
+  //Checks if the play button has been pressed 
+  {
+    //Serial.println(DDDDCLength); 
+    for (int i = 0; i < DDDDCLength; i++)
+      //Runs a for loop the amount of times as there are notes in DDDDCNotes
+    { 
+      Serial.println(i);
+      prevPlayButton = playButtonState;
+      playButtonState = digitalRead(playButton);
+      int duration = 1000 / DDDDCDurations[i];
+      int delay1 = duration + 100;
+      note(8, DDDDCNotes[i], duration);
+      //Runs our note function in order to play a note 
+      delay(delay1);
+      lcdDo(2);
+      //Updates the LCD 
+      if (playButtonState and!prevPlayButton)
+      //Edge conditions to make sure the user doesn't immediatly pause and then unpause the song 
+      {
+        counter2++;
+        lcdDo(4);
+        while (counter2 % 2 == 1)
+        //While loop Pauses the song 
+        {
+          prevPlayButton = playButtonState;
+          playButtonState = digitalRead(playButton);
+          if (playButtonState && !prevPlayButton) {
+            counter2++;
+            lcdDo(1);
+            break;
+            //Unpauses the song by leaving the while loop 
+          }
+        }
+        prevPlayButton = playButtonState;
+        playButtonState = digitalRead(playButton);
+      }
+    }
+  }
+}
+}
+void note(int a, int b, int c)
+//note playing function 
+{
+  tone(a, b, c);
+}
+void lcdDo(int x)
+//LCD displaying functions 
+{
+  switch (x) {
+  case 1:
+    lcd.clear();
+    lcd.backlight();
+    lcd.setCursor(0, 0);
```

Comment:
These steps are the same for all cases, so they don't need to be repeated. Pull them out of the switch statement.